uefi/firmwarevolume.go

```diff
@@ -74,26 +76,22 @@ func (fv FirmwareVolume) Summary() string {
 		guidString = "<invalid GUID>"
 		guidName = "Unknown"
 	}
-	return fmt.Sprintf("FirmwareVolume{\n"+
-		"    FileSystemGUID=%s (%v)\n"+
-		"    Length=%v\n"+
-		"    Signature=0x%08x\n"+
-		"    AttrMask=0x%02x\n"+
-		"    HeaderLen=%v\n"+
-		"    Checksum=0x%04x\n"+
-		"    Revision=%v\n"+
-		"    Blocks=%v\n"+
-		"}",
-		guidString, guidName,
-		fv.Length, fv.Signature, fv.AttrMask,
-		fv.HeaderLen, fv.Checksum, fv.Revision,
-		fv.Blocks,
-	)
+	return json.Marshal(struct {
```

Comment:
type-declare this struct, so you can also unmarshal

Reply:
why not add stuff like the name to FirmwareVolume?

Reply:
I could do that. I should differentiate between what's actually in the binary and what isn't now.